mariamihaelafainarea/assignment2/LowestAncestor

```diff
@@ -0,0 +1,100 @@
+    /*
+     * To change this license header, choose License Headers in Project Properties.
+     * To change this template file, choose Tools | Templates
+     * and open the template in the editor.
+     */
+
+    /**
+     *
+     * @author maria
+     */
+    public class LowestAncestor<T> {
+
+        Node first;
+        //these parameters help me find
+        //if the two values searched were found
+
+        public class Node {
+            private T key;
+            private Node left,right;
+
+            public Node getRight() {
+                return right;
+            }
+
+            public void setLeft(Node left) {
+                this.left = left;
+            }
+
+            public void setKey(T key) {
+                this.key = key;
+            }
+
+            public void setRight(Node right) {
+                this.right = right;
+            }
+
+            public Node getLeft() {
+                return left;
+            }
+
+            public T getKey() {
+                return key;
+            }
+
+            public Node(T key, Node left, Node right) {
+                this.key = key;
+                this.left = left;
+                this.right = right;
+            }
+        }
+        private Node findLowestAncestor(T x, Node root,T y,AtomicBoolean params1,AtomicBoolean params2){
+            //it is a recursive searching
+            //of the x,y.
+            if (root == null) {
+                return null;
+            }
+            if (root.key.equals(x)) {
+                params1.set(true);
+            }
+            if (root.key.equals(y)) {
+                params2.set(true);
+            }
+            Node left = findLowestAncestor(x, root.left, y,params1,params2); 
+            Node right = findLowestAncestor(x, root.right, y,params1,params2); 
+            if(root.key.equals(x) || root.key.equals(y)){
+                return root;
+            }
+            //if here are both x and y propagate root.
+            if(left != null && right != null){
+                return root;
+            }
+            //propagate what is not null
+            if (left != null) {
+                return left;
+            }
+            else if (right != null) {
+               return right;
+           }
+            return null;
+        }
+        public Node isARealAncestor(T x,T y) {
```

Comment:
This method name is not very clear. It would be better to just call it findLowestCommonAncestor or lowestCommonAncestor or something like that. And the other helper method can be called something helper. This current name suggests to me that this method will return a boolean whether a node is a real ancestor